src/hotspot/share/gc/shared/satbMarkQueue.hpp

```diff
@@ -138,9 +150,11 @@ class SATBMarkQueueSet: public PtrQueueSet {
   // buffer; the leading entries may be excluded due to filtering.
   bool apply_closure_to_completed_buffer(SATBBufferClosure* cl);
 
+  // When active, add obj to queue by calling enqueue_known_active.
   void enqueue(SATBMarkQueue& queue, oop obj) {
     if (queue.is_active()) enqueue_known_active(queue, obj);
```

Comment:
Maybe something for a different CR and probably pre-existing: `enqueue()` in some contexts is already called with `queue.is_active() == true`. I.e. this check is superfluous, particularly when called for object arrays.

Reply:
I was planning to look at that later.  I filed a reminder: https://bugs.openjdk.java.net/browse/JDK-8258607